src/external/electrs.ts

```diff
@@ -224,11 +237,33 @@ export class DefaultElectrsAPI implements ElectrsAPI {
         return amount;
     }
 
+    async getEarliestPaymentToRecipientAddressTxId(recipientAddress: string, amount?: BitcoinAmount): Promise<string> {
+        try {
+            const txes = await this.getData(this.addressApi.getAddressTxHistory(recipientAddress));
+            if (txes.length >= 25) {
```

Comment:
Hmm actually we may as well iterate over the past transactions to get the first.

Reply:
Only issue is that the paging is kinda meh:
 - the first call returns the newest 25 confirmed txs, and then up to 50 mempool txs -> meaning have to filter for confirmed txs to know if we need paging
 - if there's exactly 25 txs, from the docs it sounds like this will return all 25 and then the next call will return []

Can definitely do it if it's necessary but given that we never expect to see it, not sure whether the added complexity is worth it.